File: 0069-sqrtx/0069-sqrtx.py
class Solution(object):
    def mySqrt(self,x):
        if x < 2:
            return x  

        L, R = 2, x // 2
        while L <= R:
            M = (L + R) // 2
            M_square = M * M

            if M_square == x:
                return M
            elif M_square < x:
                L = M + 1
            else:
                R = M - 1
        return R

File: 0069-sqrtx/0069-sqrtx.java
class Solution 
{
    public int mySqrt(int x) 
    {
        if (x<2)
            return x;
        int left  = 2;
        int right = x/2;
        while (left <= right)
        {
            int mid = (left + right)/2;
            if ((long)mid*mid > x)
                right = mid - 1;
            else if ((long)mid*mid < x)
                left = mid + 1;
            else
                return mid;
        }
        return right;
    }
}